### PR TITLE
FEV-660 - Player V7| live 2.0.4| - Low network conditions - Live does not return to the correct time after low network conditions are removed

### PR DESCRIPTION
**Pull request by semarche-kaltura**
1) optimization: remove listener and handle 'timedmetadata' event in decorator
2) optimization: update live-tag on 'seeking' event like player.js does
3) prevent unnecessary re-render of custom live-tag

### CODE REVIEW

File: src/decorator/live-decorator.ts [hunk 1]
```diff
@@ -31,6 +33,16 @@ export class KalturaLiveEngineDecorator implements KalturaPlayerTypes.IEngineDec
     }
 
     dispatchEvent(event: any): any {
+        if (event.type === EventTypes.METADATA) {
+            this._plugin.handleTimedMetadata(event);
+        }
+        if (event.type === EventTypes.SEEKING) {
```
Comment (yairans) [inline src/decorator/live-decorator.ts:39]: Change it to `SEEKED` and get rid the timeout

File: src/kaltura-live-plugin.tsx [hunk 2]
```diff
@@ -132,19 +138,14 @@ export class KalturaLivePlugin
...
-            );
             this._player.configure({
                 plugins: { kava: { tamperAnalyticsHandler: this._tamperAnalyticsHandler } }
             });
             this.updateLiveStatus();
         }
     };
 
-    private _handleTimedMetadata = (e: any) => {
-        this.updateLiveTag();
```
Comment (yairans) [inline src/kaltura-live-plugin.tsx:147]: why it removed?

**Review by yairans — CHANGES_REQUESTED** [review]

File: src/kaltura-live-plugin.tsx (hunk 2 above)
Comment (semarche-kaltura) [reply src/kaltura-live-plugin.tsx:147]: according code from playkig-js the isOnLive property updated only on SEEKING event, so it does not have sense check isOnLive property after TimedMetadata comes
https://github.com/kaltura/playkit-js/blob/master/src/player.js#L1696-L1700

Comment (semarche-kaltura) [reply src/kaltura-live-plugin.tsx:147]: so new implementation updates Live tag only when `SEEKED` event happened

File: src/decorator/live-decorator.ts (hunk 1 above)
Comment (semarche-kaltura) [reply src/decorator/live-decorator.ts]: awesome hint !! 👍 💯 
updated

File: src/decorator/live-decorator.ts [hunk 3]
```diff
@@ -31,6 +33,14 @@ export class KalturaLiveEngineDecorator implements KalturaPlayerTypes.IEngineDec
     }
 
     dispatchEvent(event: any): any {
+        if (event.type === EventTypes.METADATA) {
+            this._plugin.handleTimedMetadata(event);
+        }
+        if (event.type === EventTypes.SEEKED) {
+            // check _isOnLiveEdge property after 'SEEKING' event happened
```
Comment (yairans) [inline src/decorator/live-decorator.ts:40]: the comment is irrelevant any more no?

Comment (semarche-kaltura) [reply src/decorator/live-decorator.ts:40]: relevant
after "SEEKING" happenED we check the state. It more like explanation how the plugin communicate with player.
Might be content of comment can be changed, but the goal - should be same

Comment (semarche-kaltura) [reply src/decorator/live-decorator.ts:40]: If you want I can change the comment to:
`// check _isOnLiveEdge property each time when 'SEEKED' event happens`

Comment (yairans) [reply src/decorator/live-decorator.ts:40]: I ok with any comments. just thought it was related to the timeout.

**Review by yairans — APPROVED** [review]

**Pull request merged**